### DISCUSSION TITLE
TypeError: an integer is required (got type str)

### DISCUSSION THREAD

**Discussion by ebuitragod** [apache/airflow#29507] [General]
### Apache Airflow version

2.5.1

### What happened

When I try the code in staging it does not work, although when I do it in local it does. It is a variable.get problem in which it breaks when I try to add new variables. 

here is staging:
<img width="981" alt="image" src="https://user-images.githubusercontent.com/29654051/218454952-d99bc68c-b416-4c43-8eda-52131838d65e.png">

here is in my local
<img width="1406" alt="image" src="https://user-images.githubusercontent.com/29654051/218455081-47465dd5-2e67-4873-b1c2-5f871b67d584.png">

Furthermore, it is strange that they print, but don't work:

{
  "timezones":[
    {
      "internal_emails":[
        "esperanza@agranimo.com"
      ],
      "name":"America/Santiago",
      "report_to_clients":false,
      "manual": true,
      "start_date": "23-01-30"
    },
...
  ]
}

The excercise in within a DAG factory. 

### Operating System

os

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
This is at most a discussion not an airflow issue. First of all it is Airflow 2.0.1 (as seen from the output) and not 2.5.1 as you chose. Airflow 2.0.1 is > 2 years old and there were about few thousands of bug fixes implemented since 2.0.1 and many of them around serialization so your best option is to upgrade to latest version and see if it is happening there.

Once you do it - if this still occurs, you will need to provide more details - DAGs, more logs describe circumstances.
